docs/FeatureManagement/Samples/TargetingFilter.sample.json

```diff
@@ -0,0 +1,60 @@
+{
+    "feature_management": {
+        "feature_flags": [
+            {
+                "id": "ComplexTargeting",
```

Comment:
I suggest we add a separated feature flag for each targeting evaluation way.

Reply:
I'm not sure what you mean by "evaluation way". Do you mean the different parameters? I'm fine with that, but we also need a full one, as we need to make sure everything works, not just individual parts.

At this point I don't want to add any more tests to this PR. Once it gets merged we can start to add more. The more tests added in this PR will just make it longer to approve.

Reply:
> Do you mean the different parameters?

Yes. Sure, we need to have a full one. I mean we can have feature flags: GroupTargetingFeatureFlag, UserTargetingFeatureFlag, DefaultRolloutFeatureFlag, ....

These flags are used to test the most simple and basic scenarios.

Reply:
> At this point I don't want to add any more tests to this PR. Once it gets merged we can start to add more. The more tests added in this PR will just make it longer to approve.

If our plan is to keep working on this test suite and there will be more PRs for it, I agreed that we can keep it simple for now.

Reply:
Think of this as V1, if you really look at this, we are mostly just doing positive test cases at the moment. I will probably add the negative test cases when I add variants.